### PR TITLE
Fix: correct the early return error when save_epochs=1 and optimize the clean up way

### PR DESCRIPTION
**Pull request by xiu-cs**
- Fixed an issue causing an incorrect early return when save_epochs=1.
- clean up old snapshots if the number of existing snapshots exceeds the maximum allowed (executed at the end of the process).

Special thanks to @n-poulsen  for your invaluable guidance! 🙏

### CODE REVIEW

File: deeplabcut/pose_estimation_pytorch/runners/snapshots.py [hunk 1]
```diff
@@ -113,41 +113,43 @@ def update(self, epoch: int, state_dict: dict, last: bool = False) -> None:
         ):
             current_best = self.best()
             self._best_metric = metrics[self._key]
+            
+            # Save the new best model
             save_path = self.snapshot_path(epoch, best=True)
             parsed_state_dict = {
-                k: v
+                k: v
```
Comment (n-poulsen) [inline deeplabcut/pose_estimation_pytorch/runners/snapshots.py]: You should run [black](https://github.com/psf/black) on the changes you made (with `black deeplabcut/pose_estimation_pytorch/runners/snapshots.py`) to lint the code (so there aren't these extra spaces)

File: deeplabcut/pose_estimation_pytorch/runners/snapshots.py [hunk 2]
```diff
@@ -113,41 +113,43 @@ def update(self, epoch: int, state_dict: dict, last: bool = False) -> None:
...
+            # Save regular snapshot if needed
+            should_save = last or epoch % self.save_epochs == 0
+            if should_save:
+                save_path = self.snapshot_path(epoch=epoch)
+                parsed_state_dict = {
+                    k: v
+                    for k, v in state_dict.items()
+                    if self.save_optimizer_state or k != "optimizer"
+                }
+                torch.save(parsed_state_dict, save_path)
```
Comment (n-poulsen) [inline deeplabcut/pose_estimation_pytorch/runners/snapshots.py]: I think we can be more concise here:

```suggestion
        elif last or epoch % self.save_epochs == 0:
            # Save regular snapshot if needed
            save_path = self.snapshot_path(epoch=epoch)
            parsed_state_dict = {
                k: v
                for k, v in state_dict.items()
                if self.save_optimizer_state or k != "optimizer"
            }
            torch.save(parsed_state_dict, save_path)
```

**Review by n-poulsen — CHANGES_REQUESTED** [review]
Awesome thank you @xiu-cs! Just requested a few minor changes and some linting :)

File: deeplabcut/pose_estimation_pytorch/runners/snapshots.py (hunk 1 above)
Comment (xiu-cs) [reply deeplabcut/pose_estimation_pytorch/runners/snapshots.py]: Thanks! Next time I will run black before I submit the changes.

image

**Review by n-poulsen — APPROVED** [review]
Looks good to me, thanks @xiu-cs!

**Pull request merged**